sklearn/utils/estimator_checks.py

```diff
@@ -396,6 +424,21 @@ def parametrize_with_checks(estimators):
            Passing a class was deprecated in version 0.23, and support for
            classes was removed in 0.24. Pass an instance instead.
 
+    strict_mode : bool, default=True
+        If True, the full check suite is run.
+        If False, only the non-strict part of the check suite is run.
+
+        In non-strict mode, some checks will be easier to pass: e.g., they
+        will only make sure an error is raised instead of also checking the
+        full error message.
+        Some checks are considered completely strict, in which case they are
+        treated as if they were in the estimators' `_xfails_checks` tag: they
+        will be marked as `xfail` for pytest. See :ref:`estimator_tags` for
```

Comment:
When the estimator developer has said they have no intention of their estimator passing these strict checks, why should we give them an XFAIL? Shouldn't they just not be tests?

Reply:
we discussed this a bit with @rth and @thomasjpfan , they might have input on this.

It's *much* easier implem-wise to stay consistent between `strict_mode` and the `_xfail_checks` tag. It is also much less confusing IMO.

Also, marking these checks as Xfail will inform developers with an XPASS if they actually pass, so that might be useful info.